source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -434,6 +445,22 @@ ReadOrParseState Filter::readProxyHeader(Network::IoHandle& io_handle) {
     if (nread < 1) {
       ENVOY_LOG(debug, "failed to read proxy protocol (no bytes read)");
       return ReadOrParseState::Error;
+    } else if (config_.get()->allowRequestsWithoutProxyProtocol() && buf_off_ == 0) {
```

Comment:
Your handling decides to skip the filter if not enough data is read; that's not correct. Everytime some data comes in, look at what we have, and determine whether this is defintely proxy proto v1, definately v2, definitely neither, or not-yet-determined (needs more data to decide).

/wait

Reply:
Sounds like we are saying the same case https://github.com/envoyproxy/envoy/pull/18951/files#r747243218